app/models/spree/image.rb

```diff
@@ -46,5 +46,27 @@ def no_attachment_errors
         false
       end
     end
+
+    # Spree stores attachent definitions in JSON. This converts the style name and format to
+    # strings. However, when paperclip encounters these, it doesn't recognise the format.
+    # Here we solve that problem by converting format and style name to symbols.
+    # See also: ImageSettingsController decorator.
+    #
+    # eg. {'mini' => ['48x48>', 'png']} is converted to {mini: ['48x48>', :png]}
+    def self.format_styles(styles)
+      styles_a = styles.map do |name, style|
+        style[1] = style[1].to_sym if style.is_a? Array
+        [name.to_sym, style]
+      end
+
+      Hash[styles_a]
+    end
+
+    def self.reformat_styles
+      Spree::Image.attachment_definitions[:attachment][:styles] =
+        format_styles(Spree::Image.attachment_definitions[:attachment][:styles])
+    end
+
+    reformat_styles
```

Comment:
I think this would be better off in an initializer. Not only is weird in here but I bet it gets unnecessarily evaluated more than once, at least in dev.

Reply:
this is existing code https://github.com/openfoodfoundation/openfoodnetwork/blob/7cb49ac9a8206845c935de4fe1fb38e6b197716d/app/models/spree/image_decorator.rb#L22

I dont want to fix all the spree/decorators mess in these PRs to reduce risk. Are you ok if I create a tech debt issue for this one?

Reply:
ok, thanks.
here: https://github.com/openfoodfoundation/openfoodnetwork/issues/5973